69.md

```diff
@@ -0,0 +1,126 @@
+# Zap Poll event
+
+`draft` `optional` `author:toadlyBroodle`
+
+A zap poll note is a [nostr event](01.md) (kind `6969`) for conducting paid polls—herein referred to simply as 'polls'. A poll presents two or more voting options, which participants my vote on by sending regular [zap events](57.md) which include an additional `poll_option` vote tag. Polls may include multiple recipients which participants may choose from when zapping their votes. Polls may specify `value_maximum` and `value_minimum` satoshi valuations for determining which zaps are included in the tally. Polls may specify a `consensus_threshold` for assessing the state of consensus. Polls should specify a `closed_at` time, after which results be unblinded, closed to new votes, and the tally considered final.
+
+## Purpose
+
+The purpose of poll notes is to conduct quantitative public opinion polls over nostr by requiring voters pay to participate. By tying results to real satoshi valuations, nostr polls intend to provide superior signal compared to other free polling models. Imposing real monetary costs for participation should also discourage undesired attempts to sway results.
+
+Pollers may specify multiple `p` keys, to allow participants to choose which recipient they zap, regardless of their vote choice. The option for such distribution of funds should also incentivize participation by increasing voter optionality, poller authenticity, and legitimacy of results by mitigating certain unauthentic attack vectors.
+
+By including a `value_maximum` limit, polls can stop single 'whale' votes discounting many smaller 'shrimp' votes. Likewise, by including a `value_minimum` limit, polls can make automated low-value vote flooding attacks prohibitively expensive. However, both limits remain optional to allow for freedom in poll design.
+
+By setting `value_maximum` and `value_minimum` equal, a more traditional style poll may be designed, which weights each vote equal and limits each participant to a single vote.
+
+The optional `consensus_threshold` is intended as a simple 'measuring bar', defined as the minimum percentage for any single `poll_option`'s percentage of the total tally to attain poll consensus status.
+
+A careful balancing of all poll attributes should enable pollers to conduct tailored polls that deliver meaningful and valuable outcomes.
+
+## Zap poll format
+
+A poll event:
+* MUST specify 1 or more `p` tags, each including the same primary hosting relay
+* if an `e` tag is specified, it:
+  * MUST include the same primary hosting relay URL as the `p` tags
+* MUST contain a primary description string, specified in the `content` field
+* MUST contain at least 2 `poll_option` tags, each specifying an index and a unique description string, formatted as below
```

Comment:
Must the indices be numeric? Must they be in order? If so, could they be omitted and inferred from the order they appear in the tag list?

What if an index is repeated? Would this invalidate the poll? Is it undefined behavior?

Reply:
Indices must be numeric, must be present, but don't need to be in order, as they are strings. Clients can decide how to handle non-compliant, undefined, event structures.
Defined structure:
>["poll_option", "0", "poll option 0 description string"],
>["poll_option", "1", "poll option 1 description string"],
>["poll_option", "n", "poll option <n> description string"]